### DISCUSSION TITLE
Support custom views in List Resource

### DISCUSSION THREAD

**Discussion by viezel** [filamentphp/filament#84] [Help] [answered]
When creating a unique experience for the admins, you often need extra content. Metrics, guidance, banners, whatever that make sense in a given app. I have 2 suggestions for this:

### Suggestion 1

Adding 2 slots to the List Resource view to be able to support custom views. Something like this

```
custom header view

TABLE view

custom footer  view
```

```php
class CustomerResource extends Resource
{
    public static function headerView()
    {
        return view('admin.customers.header');
    }

    public static function footerView()
    {
        return view('admin.customers.footer');
    }
}
```

### Suggestion 2

Override the view, and do a callout to the tableview

```php
class CustomerResource extends Resource
{
    public static function tableView()
    {
        return view('admin.customers.table');
    }
}
```

resources/views/admin/customers/table.blade.php

```blade

<div>my custom header</div>

@include('filament::resources.pages.list-records')

<div>my custom footer</div>

```

**Comment by danharrin** [comment, marked as answer]
This is already possible by overriding the `$view` on the page component with a custom view - the code to render a table is very minimal.

It would be nice to have a header and footer view, you're right 👍 If you wanted to PR that it'd be great.

**Reply by viezel** [reply]
Yes, didnt think of that. This works great. No need for extra. 

```php
class ListCustomers extends ListRecords
{
    public static $resource = CustomerResource::class;

    public static $view = 'admin.customers.table';
}
```

**Reply by viezel** [reply]
@danharrin when I override the $view property, then the livewire interactive logic is broken. I cannot search or filter the table.

**Reply by danharrin** [reply]
> @danharrin when I override the $view property, then the livewire interactive logic is broken. I cannot search or filter the table. 

Can you post the code in your broken view please?
